core/src/gas.rs

```diff
@@ -101,10 +104,28 @@ impl GasCounter {
         }
     }
 
+    /// Increase gas by `amount`.
+    ///
+    /// Called when gas unreservation is occurred.
+    // We don't decrease `burn` counter because `GasTree` manipulation is handled by separated function
+    pub fn increase(&mut self, amount: u64) -> bool {
+        match self.left.checked_add(amount) {
+            None => false,
+            Some(new_left) => {
+                self.left = new_left;
+                true
+            }
+        }
+    }
+
```

Comment:
Could you, please, comment that method to prevent it's usage for nearest future?

Due to the fact that we plan add ability to unreserve gas into current message handling that's really unstable and shouldn't be public atm!